src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs

```diff
@@ -106,24 +106,8 @@ public Test BuildFrom(FixtureMethod method, Test suite)
             List<ITestBuilder> builders = new List<ITestBuilder>(
                 method.Method.GetAttributes<ITestBuilder>(false));
 
-            // See if we need a CombinatorialAttribute added
-            bool needCombinatorial = true;
-            foreach (var attr in builders)
-            {
-                if (attr is CombiningStrategyAttribute)
-                    needCombinatorial = false;
-            }
-
-            // This check must be done before CombinatorialAttribute gets added to the builders collection
-            var hasBuildersSpecified = builders.Count > 0;
-
-            // We could check to see if here are any data attributes specified
-            // on the parameters but that's what CombinatorialAttribute does
-            // and it simply won't return any cases if it finds nothing.
-            // TODO: We need to add some other ITestBuilder than a combinatorial attribute
-            // because we want the attribute to generate an error if it's present on
-            // a generic method.
-            if (needCombinatorial)
+            // See if we need a CombinatorialAttribute added for parametarized data
+            if (MethodHasDataAttributeParameters(method) && !HasCombinatorialStrategy(builders))
```

Comment:
I would have put the loop for HasCombinatorialStrategy inline here, but that's a matter of style. Alternatively, a Linq expression would be OK here in my view, since it would be a very simple one.

[You asked about the standard earlier... my own general rule is that the code has to be understandable by __the team that you have__! IOW, there's no general principle you can apply without knowing who the developers are.]